### PR TITLE
Adding "latency" var to GatewayClient

### PR DESCRIPTION
**Pull request by ThatGuyJustin**
+ Latency (in seconds) to the gateway

### CODE REVIEW

**Comment by A5rocks** [comment]
`time.perf_counter()` is raising errors in python 2. Maybe detect if the current version is python 2 using `six`, then if so, alias `time.clock` as `time.perf_counter`. Just an idea.

https://travis-ci.org/b1naryth1ef/disco/jobs/615317839#L743

File: disco/gateway/client.py [hunk 1]
```diff
@@ -71,6 +72,11 @@ def __init__(self, client, max_reconnects=5, encoder='json', zlib_stream_enabled
         self._heartbeat_task = None
         self._heartbeat_acknowledged = True
 
+        # Latency
+        self._last_send = time.perf_counter()
+        self._last_ack = time.perf_counter()
+        self.latency = float('inf')
```
Comment (b1naryth1ef) [inline disco/gateway/client.py]: `self.latency = -1`

File: disco/gateway/client.py [hunk 2]
```diff
@@ -90,11 +96,18 @@ def heartbeat_task(self, interval):
                 self._heartbeat_acknowledged = True
                 self.ws.close(status=4000)
                 return
+            else:
```
Comment (b1naryth1ef) [inline disco/gateway/client.py]: Delete this else, just `self._last_heartbeat = time.time()`

File: disco/gateway/client.py [hunk 3]
```diff
@@ -71,6 +72,11 @@ def __init__(self, client, max_reconnects=5, encoder='json', zlib_stream_enabled
         self._heartbeat_task = None
         self._heartbeat_acknowledged = True
 
+        # Latency
+        self._last_send = time.perf_counter()
```
Comment (b1naryth1ef) [inline disco/gateway/client.py]: `self._last_heartbeat = 0`

File: disco/gateway/client.py [hunk 4]
```diff
@@ -90,11 +96,18 @@ def heartbeat_task(self, interval):
...
                 return
+            else:
+                self._last_send = time.perf_counter()
 
             self._send(OPCode.HEARTBEAT, self.seq)
             self._heartbeat_acknowledged = False
             gevent.sleep(interval / 1000)
 
+    def ack(self):
+        ack_time = time.perf_counter()
```
Comment (b1naryth1ef) [inline disco/gateway/client.py]: Delete this function.

File: disco/gateway/client.py [hunk 5]
```diff
@@ -108,6 +121,7 @@ def handle_heartbeat(self, _):
     def handle_heartbeat_acknowledge(self, _):
         self.log.debug('Received HEARTBEAT_ACK')
         self._heartbeat_acknowledged = True
+        self.ack()
```
Comment (b1naryth1ef) [inline disco/gateway/client.py]: `self.latency = int((time.time() - self._last_heartbeat) * 1000)`

**Comment by ThatGuyJustin** [comment]
All changes made and tested!

**Pull request merged**